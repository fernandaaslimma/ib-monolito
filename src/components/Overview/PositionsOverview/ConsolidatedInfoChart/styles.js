import styled from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";

export const InvestmentConsolidatedInfo = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  position: absolute;
  top: calc(50% - 34%);
  margin: 0 auto;

  z-index: 100000;
`;

export const InvestmentConsolidatedInfoSpan = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(14)};
  line-height: 125%;
  /* identical to box height, or 17px */

  text-align: center;
  letter-spacing: 0.45px;

  margin: ${rem(15)} 0 ${rem(4)} 0;
  color: #587485;
`;

export const ClickableItem = styled.span`
  cursor: pointer;
`;

export const HideValueWrapper = styled.div`
  padding-bottom: ${rem(13)};
  border-bottom: ${rem(1)} solid #d9e0e4;
`;

export const WrapperCommomIcon = styled.div`
  margin-top: ${rem(16)};
`
