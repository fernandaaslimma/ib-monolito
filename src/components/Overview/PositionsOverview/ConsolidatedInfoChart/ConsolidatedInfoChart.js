import React, { memo } from "react";
import { rem, remFontSize } from "../../../../styles/tools";
import { BRL_CURRENCY } from "../../../../utils/constants";
import { translate } from "../../../../utils/i18n";
import HideableValue from "../../../common/HideableValue";
import {
  HideValueWrapper,
  InvestmentConsolidatedInfo,
  InvestmentConsolidatedInfoSpan,
  ClickableItem,
  WrapperCommomIcon
} from "./styles";
import IconCommom from "../../../common/Icon";
import { Icon } from "react-bocombbm-components";

function ConsolidatedInfoChart({ hideValues, total, callback, currentCoin, isOffShore }) {
  return (
    <InvestmentConsolidatedInfo>
      {hideValues ? (
        <ClickableItem style={{ cursor: "pointer" }} onClick={() => callback()}>
          <Icon
            type="View"
            width={32}
            height={32}
            cursorPointer
            color="#3976CF"
          />
        </ClickableItem>
      ) : (
        <ClickableItem onClick={() => callback()}>
          <Icon
            type="HideView"
            width={32}
            height={32}
            cursorPointer
            color="#3976CF"
          />
        </ClickableItem>
      )}
      <InvestmentConsolidatedInfoSpan>
        {`${translate("TOTAL_INVESTED")}`}
      </InvestmentConsolidatedInfoSpan>
      <HideValueWrapper>
        <HideableValue
          hide={hideValues}
          currency={isOffShore ? currentCoin : BRL_CURRENCY}
          value={total.grossBalance}
          currencyColor="#99B5C6"
          currencySize={12}
          styles={`font-size: ${remFontSize(
            22
          )}; color: "#2d4758"; font-family: Lato Bold; margin-right: ${rem(
            4
          )}; display: flex; align-items: center;`}
        />
      </HideValueWrapper>
      {isOffShore && (<WrapperCommomIcon data-testid="flagIcon">
        <IconCommom width="28" height="28" type={currentCoin} />
      </WrapperCommomIcon>)}
    </InvestmentConsolidatedInfo>
  );
}

export default memo(ConsolidatedInfoChart, (prevProps, nextProps) => {
  return (
    prevProps.total.grossBalance === nextProps.total.grossBalance &&
    prevProps.hideValues === nextProps.hideValues &&
    prevProps.currentCoin === nextProps.currentCoin
  );
});

ConsolidatedInfoChart.displayName = "ConsolidatedInfoChart";
