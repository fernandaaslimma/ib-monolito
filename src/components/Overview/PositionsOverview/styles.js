import styled, { css } from "styled-components";
import { gray90, neutral200, white } from "../../../styles/settings";
import { rem } from "../../../styles/tools";

export const Wrapper = styled.section`
  background: ${white};
`;

export const Container = styled.div`
  padding: ${rem(12)} ${rem(16)} 0 ${rem(16)};
`;

export const IconWrapper = styled.div`
  cursor: pointer;
  position: absolute;
  top: calc(100% - 55%);
  z-index: 999;
  right: ${rem(4)};

  ${({ position }) => {
    if (position && position === 1) {
      return css`
        left: ${rem(4)};
        right: auto;
      `;
    }
  }};

  ${({ angle }) =>
    angle &&
    css`
      transform: rotate(${angle}deg);
    `}
`;

export const ChartController = styled.div`
  display: flex;
  justify-content: center;
  margin: ${rem(5)};
`;

export const Dot = styled.div`
  width: ${rem(6)};
  height: ${rem(6)};
  background-color: ${gray90};

  margin: ${rem(4)};
  border-radius: ${rem(10)};

  ${({ active }) =>
    active &&
    css`
      background-color: ${neutral200};
    `};
`;
