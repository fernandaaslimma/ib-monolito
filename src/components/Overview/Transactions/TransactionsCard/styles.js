import styled from "styled-components";
import { rem, remFontSize } from "../../../../styles/tools";
import {
  gray300,
  white,
  gray200,
  blue20,
  gray80
} from "../../../../styles/settings";

export const Card = styled.div`
  margin: 0px 0px ${rem(32)} 0px;
  background: ${blue20};
`;

export const CardTitle = styled.p`
  color: ${gray300};
  font-family: Lato Bold;
  padding: 0px 0px ${rem(16)} 0px;
`;

export const CardContent = styled.div`
  background: ${white};
  border: 1px solid ${gray80};
  padding: ${rem(8)} ${rem(16)};
`;

export const Section = styled.div`
  display: flex;
  font-family: Lato;
  margin: ${rem(8)} 0;
`;

export const Info = styled.span`
  flex-grow: 1;
`;
export const Label = styled.p`
  font-size: ${remFontSize(12)};
  line-height: 125%;
  letter-spacing: ${rem(0.452308)};
  color: ${gray200};
  margin-bottom: ${rem(4)};
`;

export const Value = styled.span`
  font-weight: bold;
  line-height: 18px;
  letter-spacing: ${rem(0.44999998807907104)};
  font-size: ${remFontSize(14)};
  color: ${gray300};
`;
