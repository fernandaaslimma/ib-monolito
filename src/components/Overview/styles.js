import styled from "styled-components";
import { rem } from "../../styles/tools";

export const MainWrapper = styled.section`
  background-color: #fff;
  max-width: ${rem(600)};
  margin: auto;
`;

export const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  justify-content: center;
  background-color: #fff;
`;

export const StickyWrapper = styled.div`
  overflow: auto;
  position: sticky;
  bottom: 0;
  background: white;
  display: flex;
  flex-direction: column;
  border-top: solid ${rem(1)} #d3dde4;
`;
