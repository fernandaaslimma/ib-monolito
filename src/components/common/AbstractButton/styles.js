import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import {
  lightBlue,
  darkBlue,
  green,
  white,
  grey,
  lightRed,
  lighterRed,
  grey70
} from "../../../styles/settings";

export const ButtonTag = styled.button`
  display: flex;
  justify-content: center;
  font-family: 'Lato Bold', 'Lato';
  background: none;
  /* border-radius: ${rem(4)}; */
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.2)};
  line-height: ${rem(45)};
  padding: ${props => (props.noPadding ? "0 rem(20)" : 0)};
  position: relative;
  text-align: center;
  text-decoration: none;
  /* box-shadow: 0 4px 7px 0 rgba(74, 144, 226, 0.3); */
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  /* opacity: ${props =>
    props.opacity && props.disabled ? props.opacity : 1}; */

  ${({ width, isPending }) =>
    width &&
    isPending &&
    css`
      min-width: ${rem(width)} !important;
      margin-right: 0 !important;
      text-align: center;
      padding: 0 ${rem(10)} !important;

      ${media.lg(css`
        margin-right: ${rem(20)} !important;
      `)};
    `};

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "100%")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    padding: 0 ${rem(28)};
    line-height: ${rem(42)};
  `)};

  ${({ noPadding }) =>
    noPadding &&
    media.lg(css`
      padding: 0;
    `)}

  :active {
    transform: translateY(${rem(1)});
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      border: solid ${rem(1)} ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(117, 214, 184, 0.39);
      color: ${white};
      font-family: "Lato Bold", Lato;

      :hover:not([disabled]) {
        background-color: ${lightBlue};
        border-color: ${lightBlue};
        box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(51, 204, 208, 0.39);
      }
    `};

  ${({ isWarning }) =>
    isWarning &&
    css`
      background-color: ${lightRed};
      border: solid ${rem(1)} ${lightRed};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(233, 103, 103, 0.3);
      color: ${white};

      :hover:not([disabled]) {
        background-color: ${lighterRed};
        border-color: ${lighterRed};
      }
    `};

  ${({ isCallToAction, isWarning }) =>
    !isCallToAction &&
    !isWarning &&
    css`
      /* border: solid ${rem(1)} ${darkBlue};
      color: ${white}; */

      :hover:not([disabled]) {
        opacity: 0.7;
      }
    `};

  ${({ small }) =>
    small &&
    css`
      font-size: ${rem(11)} !important;
      height: ${rem(40)} !important;
      line-height: ${rem(40)} !important;
    `};

  ${({ disabled, loading, opacity }) =>
    disabled &&
    !loading &&
    opacity &&
    css`
      background-color: ${!opacity && grey};
      opacity: ${opacity};
      box-shadow: none;
      border: solid ${rem(1)} ${grey};
      color: ${white};
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ disabled, loading }) =>
    disabled &&
    loading &&
    css`
      box-shadow: none;
      cursor: not-allowed;
      pointer-events: none;
    `};
  ${({ disabled, loading }) =>
    disabled &&
    !loading &&
    css`
      background-color: ${grey};
      box-shadow: none;
      border: solid ${rem(1)} ${grey};
      color: ${white};
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ disabled, loading }) =>
    disabled &&
    loading &&
    css`
      box-shadow: none;
      cursor: not-allowed;
      pointer-events: none;
    `};

  ${({ isSecondary }) =>
    isSecondary &&
    css`
      background-color: ${white};
      border-color: ${darkBlue};
      color: ${darkBlue};
      :hover:not([disabled]) {
        background-color: ${darkBlue};
        border-color: ${darkBlue};
        color: ${white};
      }
    `}

  ${({ isSpanButton }) =>
    isSpanButton &&
    css`
      background-color: transparent;
      border: unset;
      color: ${grey70};
      box-shadow: unset;

      &:hover {
        background-color: unset !important;
        color: unset !important;
        cursor: text;
      }

      &:focus {
        outline: unset !important;
      }

      &:active {
        transform: unset !important;
      }
    `};

  ${({ heightPx }) =>
    heightPx &&
    css`
      line-height: ${rem(heightPx)};
      height: ${rem(heightPx)};
    `}
`;
