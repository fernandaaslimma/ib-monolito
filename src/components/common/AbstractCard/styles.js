import styled, { css } from "styled-components";

import { white } from "../../../styles/settings";
import { rem } from "../../../styles/tools";

export const CardWrapper = styled.div`
  border-radius: ${rem(4)};
  background: ${white};
  padding: ${rem(16)} ${rem(30)};
  margin-top: ${({ marginTop }) => marginTop};
  margin-bottom: ${({ marginBottom }) => marginBottom};
  box-shadow: 0px 2px 22px 2px rgba(225, 237, 246, 0.5);
  max-width: 100%;
  position: relative;

  ${({ height }) =>
    height &&
    css`
      height: ${height};
    `};
`;
