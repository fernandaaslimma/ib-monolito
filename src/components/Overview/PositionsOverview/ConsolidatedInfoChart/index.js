import ConsolidatedInfoChart from "./ConsolidatedInfoChart";

export default ConsolidatedInfoChart;
