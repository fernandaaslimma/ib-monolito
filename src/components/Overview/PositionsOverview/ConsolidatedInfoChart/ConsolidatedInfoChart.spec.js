import { mount } from "enzyme";
import { render } from "@testing-library/react";
import React from "react";
import ConsolidatedInfoChart from "./ConsolidatedInfoChart";
import { ClickableItem } from "./styles";

const props = {
  hideValues: true,
  total: 10000,
  callback: jest.fn(),
  isOffShore: false,
  iconType: "USD"
};
const newProps = { ...props, hideValues: false };

describe("ConsolidatedInfoChart", () => {
  it("should match snapshot", () => {
    expect(mount(<ConsolidatedInfoChart {...props} />)).toMatchSnapshot();
  });

  it("should match snapshot when offShore is true", () => {
    expect(mount(<ConsolidatedInfoChart {...{ ...props, isOffShore: true}} />)).toMatchSnapshot();
  });

  it("should test rerender for React.memo", () => {
    const { rerender } = render(<ConsolidatedInfoChart {...props} />);
    rerender(<ConsolidatedInfoChart {...newProps} />);
  });

  it("should call callback with hideValues as true", () => {
    const wrapper = mount(<ConsolidatedInfoChart {...props} />);
    wrapper.find(ClickableItem).simulate("click");
    expect(props.callback).toHaveBeenCalled();
  });

  it("should call callback with hideValues as false", () => {
    const wrapper = mount(<ConsolidatedInfoChart {...newProps} />);
    wrapper.find(ClickableItem).simulate("click");
    expect(props.callback).toHaveBeenCalled();
  });

  it("should render component when offShore is true", () => {
    const wrapper = mount(<ConsolidatedInfoChart {...{ ...props, isOffShore: true}} />);
    const offSoreFlag = wrapper.find('[data-testid="flagIcon"]')
    expect(offSoreFlag.exists()).toBe(true);
  });
});
