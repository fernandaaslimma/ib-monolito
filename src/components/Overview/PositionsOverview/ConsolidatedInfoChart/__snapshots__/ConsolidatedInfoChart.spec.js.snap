// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsolidatedInfoChart should match snapshot 1`] = `
<section
  className="sc-ifAKCX faebnB"
>
  <span
    className="sc-bZQynM IAXmQ"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <span
      className="sc-cSHVUG cixzVP"
      color="#3976CF"
      height={32}
      spacing={
        Object {
          "bottom": "none",
          "left": "none",
          "right": "none",
          "top": "none",
        }
      }
      width={32}
    >
      <svg
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h20v20H0z"
          />
          <path
            d="M9.624 7.085a3.416 3.416 0 0 0-3.412 3.412 3.416 3.416 0 0 0 3.412 3.412 3.416 3.416 0 0 0 3.412-3.412 3.416 3.416 0 0 0-3.412-3.412zm0-2.085c4.39 0 7.888 4.339 8.443 5.066l.072.095c.146.2.146.472 0 .672-.154.21-3.826 5.16-8.515 5.16-4.39 0-7.888-4.338-8.443-5.065l-.072-.096a.57.57 0 0 1 0-.67C1.263 9.95 4.935 5 9.624 5zm0 1.137c-3.461 0-6.447 3.284-7.331 4.36.886 1.074 3.877 4.36 7.331 4.36 3.462 0 6.447-3.285 7.332-4.36-.886-1.075-3.878-4.36-7.332-4.36zm0 2.085a2.277 2.277 0 0 1 2.275 2.275 2.277 2.277 0 0 1-2.275 2.274 2.277 2.277 0 0 1-2.274-2.274 2.277 2.277 0 0 1 2.274-2.275z"
            fill="currentColor"
            fillRule="nonzero"
            stroke="currentColor"
            strokeWidth=".3"
          />
        </g>
      </svg>
    </span>
  </span>
  <span
    className="sc-EHOje iTLkxv"
  >
    undefined
  </span>
  <div
    className="sc-gzVnrw jMalEa"
  >
    <div
      className="sc-bdVaJa ikBUBB"
      data-test="HideableValue"
    >
      <span
        className="sc-bxivhb oIDRU"
      >
        R$
      </span>
      <span
        className="sc-bwzfXH cdemNH"
      >
        ******
      </span>
    </div>
  </div>
</section>
`;

exports[`ConsolidatedInfoChart should match snapshot when offShore is true 1`] = `
<section
  className="sc-ifAKCX faebnB"
>
  <span
    className="sc-bZQynM IAXmQ"
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <span
      className="sc-cSHVUG cixzVP"
      color="#3976CF"
      height={32}
      spacing={
        Object {
          "bottom": "none",
          "left": "none",
          "right": "none",
          "top": "none",
        }
      }
      width={32}
    >
      <svg
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M0 0h20v20H0z"
          />
          <path
            d="M9.624 7.085a3.416 3.416 0 0 0-3.412 3.412 3.416 3.416 0 0 0 3.412 3.412 3.416 3.416 0 0 0 3.412-3.412 3.416 3.416 0 0 0-3.412-3.412zm0-2.085c4.39 0 7.888 4.339 8.443 5.066l.072.095c.146.2.146.472 0 .672-.154.21-3.826 5.16-8.515 5.16-4.39 0-7.888-4.338-8.443-5.065l-.072-.096a.57.57 0 0 1 0-.67C1.263 9.95 4.935 5 9.624 5zm0 1.137c-3.461 0-6.447 3.284-7.331 4.36.886 1.074 3.877 4.36 7.331 4.36 3.462 0 6.447-3.285 7.332-4.36-.886-1.075-3.878-4.36-7.332-4.36zm0 2.085a2.277 2.277 0 0 1 2.275 2.275 2.277 2.277 0 0 1-2.275 2.274 2.277 2.277 0 0 1-2.274-2.274 2.277 2.277 0 0 1 2.274-2.275z"
            fill="currentColor"
            fillRule="nonzero"
            stroke="currentColor"
            strokeWidth=".3"
          />
        </g>
      </svg>
    </span>
  </span>
  <span
    className="sc-EHOje iTLkxv"
  >
    undefined
  </span>
  <div
    className="sc-gzVnrw jMalEa"
  >
    <div
      className="sc-bdVaJa ikBUBB"
      data-test="HideableValue"
    >
      <span
        className="sc-bwzfXH cdemNH"
      >
        ******
      </span>
    </div>
  </div>
  <div
    className="sc-htoDjs bSNJUi"
    data-testid="flagIcon"
  />
</section>
`;
